src/modules/milestoneSubModules/SpecificFundingManager.sol

```diff
@@ -0,0 +1,318 @@
+
+            return newAmount;
+        }
+    }
+
+    /// @inheritdoc ISpecificFundingManager
+    function withdrawSpecificMilestoneFunding(
+        uint milestoneId,
+        uint withdrawAmount
+    )
+        external
+        validWithdrawAmount(withdrawAmount, milestoneId)
+        validAmount(withdrawAmount)
+        returns (uint)
+    {
+        address funder = _msgSender();
+        if (milestoneIdFundingAmounts[milestoneId][funder] == withdrawAmount) {
+            withdrawAllSpecificMilestoneFunding(milestoneId);
+            return 0;
+        } else {
+            milestoneIdFundingAmounts[milestoneId][funder] -= withdrawAmount;
+            milestoneIdToFundingAddresses[milestoneId].fundingAmount -=
+                withdrawAmount;
+
+            uint newAmount = milestoneIdFundingAmounts[milestoneId][funder];
+
+            __Module_proposal.token().transfer(funder, withdrawAmount);
+
+            emit SpecificMilestoneFundingWithdrawn(
+                milestoneId, newAmount, funder
+            );
+
+            return newAmount;
+        }
+    }
+
+    function initialFundSpecificMilestone(uint milestoneId, uint addAmount)
+        private
+        returns (uint)
+    {
+        address funder = _msgSender();
+        milestoneIdToFundingAddresses[milestoneId].funders.push(funder); //@note Not checking for duplicates/ might wanna test this
+        milestoneIdToFundingAddresses[milestoneId].fundingAmount += addAmount;
+
+        milestoneIdFundingAmounts[milestoneId][funder] = addAmount;
+
+        __Module_proposal.token().transferFrom(funder, address(this), addAmount);
+
+        emit SpecificMilestoneFunded(milestoneId, addAmount, funder);
+
+        return addAmount;
+    }
+
+    function withdrawAllSpecificMilestoneFunding(uint milestoneId) private {
+        address funder = _msgSender();
+
+        uint withdrawAmount = milestoneIdFundingAmounts[milestoneId][funder];
+
+        milestoneIdFundingAmounts[milestoneId][funder] = 0;
+        milestoneIdToFundingAddresses[milestoneId].fundingAmount -=
+            withdrawAmount;
+
+        removeFunder(milestoneId, funder);
+
+        __Module_proposal.token().transfer(funder, withdrawAmount);
+
+        emit SpecificMilestoneFundingRemoved(milestoneId, funder);
+    }
+
+    //----------------------------------
+    // Collect Funding Functions
+
+    /// @inheritdoc ISpecificFundingManager
+    function collectFunding(uint milestoneId, uint amountNeeded)
+        external
+        onlyMilestoneManagerAccess
+        fundingNotCollected(milestoneId)
+        validAmount(amountNeeded)
+        returns (uint)
+    {
+        milestoneIdToFundingAddresses[milestoneId].fundingCollected = true;
+
+        address[] memory funders =
+            milestoneIdToFundingAddresses[milestoneId].funders;
+
+        uint length = funders.length;
+
+        //If we dont have any funders stop the collecting
+        if (length == 0) {
```

Comment:
validAmount modifier is checked to not be zero, therefore it has to have some funders ?
making if section 242 - 245 redundant.